server/app/controllers/admin/AdminProgramBlockQuestionsController.java

```diff
@@ -144,4 +145,33 @@ public Result setOptional(
     return redirect(
         controllers.admin.routes.AdminProgramBlocksController.edit(programId, blockDefinitionId));
   }
+
+  /** POST endpoint for changing position of a question in its block. */
+  @Secure(authorizers = Labels.CIVIFORM_ADMIN)
+  public Result move(
+      Request request, long programId, long blockDefinitionId, long questionDefinitionId)
+      throws InvalidQuestionPositionException {
```

Comment:
suggestion (blocking): if I understand correctly, this error could be thrown as a result of bad user input. If so, handle the error here and return `badRequest()` instead of throwing and causing a 500

Reply:
Done.